torch/optim/sgd.py

```diff
@@ -3,31 +3,37 @@
 
 class SGD(Optimizer):
     """Implements stochastic gradient descent (optionally with momentum).
-
     Args:
         params (iterable): iterable of parameters to optimize or dicts defining
             parameter groups
         lr (float): learning rate
         momentum (float, optional): momentum factor (default: 0)
         weight_decay (float, optional): weight decay (L2 penalty) (default: 0)
+        dampening (float, optional): dampening for momentum (default: 0)
+        nesterov(bool, optional): enables Nesterov momentum (default: False)
 
     Example:
-
         >>> optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9)
         >>> optimizer.zero_grad()
         >>> loss_fn(model(input), target).backward()
         >>> optimizer.step()
     """
 
     def __init__(self, params, lr=required, momentum=0, dampening=0,
-                 weight_decay=0):
+                 weight_decay=0, nesterov=False):
         defaults = dict(lr=lr, momentum=momentum, dampening=dampening,
-                        weight_decay=weight_decay)
+                        weight_decay=weight_decay, nesterov=nesterov)
+        if nesterov and (momentum <= 0 and dampening != 0):
```

Comment:
I pulled this one straight from [the legacy version](https://github.com/pytorch/pytorch/blob/master/torch/legacy/optim/sgd.py#L43), but I think you're correct that it should be 0 so that the error is raised if either momentum is non-positive **or** dampening is non-zero, yeah.